### DISCUSSION TITLE
Detailed inpainting technique

### DISCUSSION THREAD

**Discussion by johnbrownlow** [Stability-AI/StableSwarmUI#319] [Q&A] [answered]
I'm really struggling with the inpainting. Can someone step-by-step me through how I can inpaint a small distorted face? I want to blow up a section of the picture, mask it off, inpaint it a a higher resolution, then zoom back out. However in the current interface I don't see how to achieve that. Also can someone be very explicit about how the white and black paints work exactly, and what the opacity does in a mask layer?

Also the fact that the default extra layer is a paint layer not a mask layer is extraordinarily confusing for new users, as is the fact that mask and paint layers look exactly the same.

**Comment by mcmonkey4eva** [comment]
The image editor is very much incomplete, sorry.

The easiest way to do that right now is skip the image editor and just generate with a prompt of `<segment:face> a beautiful face yadda yadda`, which will automatically detect the face and do all the other bits for you

Re how masking works: it's explained in the Mask parameter doc:
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/5dcd404d-6c6c-46e5-bde5-b212ce0bfc5e)

**Comment by johnbrownlow** [comment]
Thank you for for the reponse! It would not have occurred to me to try that!

**Comment by mcmonkey4eva** [comment, marked as answer]
_Also_, for when using regular masks / image editor masks, I added `Mask Shrink Grow` which performs the automatic crop-to-mask-and-then-grow-again-after behavior

![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/fc0e893a-ec0d-4de3-9244-ec903fc3335e)

(There's a long list of general improvements to the image editor interface still to come)

**Comment by arafatx** [comment]
I see the UI is getting better and better. When the inpainting feature is completed with 'masked only' inpainting (no comfy backend thing), it would be a great time to stay away from Automatic1111

**Reply by mcmonkey4eva** [reply]
What do you mean? Isn't "masked only" in auto the same as the "mask shrink-grow" described above?

**Reply by arafatx** [reply]
Okay, my bad. I think I might have skipped the answer earlier when you mentioned that `the image editor is very much incomplete`. I'll give that one a try. Thank you
